### PR TITLE
feat: implement cards for the blog

### PR DESCRIPTION
**Pull request by mabsattar**
# Description

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: apps/website/blog/2024-02-28-maci-v1.2.0.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ authors:
   url: https://pse.dev
   image_url: /img/pse-logo-round.png
 tags: [release, audit, security, documentation]
+excerpt: "We are pleased to announce the release of MACI [v1.2.0](https://github.com/privacy-scaling-explorations/maci/releases/tag/v1.2.0)!"
```
Comment (ctrlc03) [inline apps/website/blog/2024-02-28-maci-v1.2.0.md:10]: given the markdown is not rendering in the card preview, can remove links and just keep v1.2.0

File: apps/website/blog/2024-05-08-ethdam.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ authors:
   url: https://x.com/ctrlc03
   image_url: https://avatars.githubusercontent.com/u/93448202?v=4
 tags: [security, bug, babyjubjub, input validation]
+excerpt: "During [ETHDam's Quadratic Funding round](https://ethdamqf.com/#/), run on [clr.fund](https://clr.fund), we discovered a critical bug in MACI. The issue stemmed from the lack of validation on MACI public keys within the [Poll contract](/docs/developers-references/smart-contracts/solidity-docs/Poll). A user (spoiler alert, it was a self-inflicted denial of service (DoS)) was able to submit a MACI public key which was not a point of the Baby JubJub elliptic curve, and it broke everything."
```
Comment (ctrlc03) [inline apps/website/blog/2024-05-08-ethdam.md:11]: given the markdown is not rendering in the card preview, can remove links

File: apps/website/blog/2024-07-23-q2-review.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ authors:
   url: https://x.com/ctrlc03
   image_url: https://avatars.githubusercontent.com/u/93448202?v=4
 tags: [voting, security, anonymity, roadmap, grants]
+excerpt: "Glad to have you here! It's that time of the year where we have a few MACI roadmap updates to share. We’re going to take a moment to look at what we accomplished in Q2, as the [most active](https://next.ossinsight.io/analyze/privacy-scaling-explorations?period=past_12_months#overview) project in the entire PSE GitHub org."
```
Comment (ctrlc03) [inline apps/website/blog/2024-07-23-q2-review.md:11]: let's remove links

File: apps/website/blog/2024-02-28-maci-v1.2.0.md (hunk 1 above)
Comment (mabsattar) [reply apps/website/blog/2024-02-28-maci-v1.2.0.md]: done

File: apps/website/blog/2024-05-08-ethdam.md (hunk 2 above)
Comment (mabsattar) [reply apps/website/blog/2024-05-08-ethdam.md]: done

File: apps/website/blog/2024-07-23-q2-review.md (hunk 3 above)
Comment (mabsattar) [reply apps/website/blog/2024-07-23-q2-review.md]: its done!

File: apps/website/blog/2024-02-28-maci-v1.2.0.md (hunk 1 above)
Comment (ctrlc03) [reply apps/website/blog/2024-02-28-maci-v1.2.0.md]: looks like this one has [ ] around v1.2.0 still

File: apps/website/src/pages/blogs.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
```
Comment (ctrlc03) [inline apps/website/src/pages/blogs.tsx:1]: please can you fix the linter errors here? you can run it locally with `pnpn lint:ts:fix`

File: apps/website/blog/2024-02-28-maci-v1.2.0.md (hunk 1 above)
Comment (mabsattar) [reply apps/website/blog/2024-02-28-maci-v1.2.0.md]: oh okay, lemme check that.

Comment (mabsattar) [reply apps/website/blog/2024-02-28-maci-v1.2.0.md]: its done

**Review by ctrlc03 — APPROVED** [review]
Thank you so much!!

**Pull request merged**